lib/pages/contacts_tab/contacts_tab_body_view.dart

```diff
@@ -20,88 +18,74 @@ class ContactsTabBodyView extends StatelessWidget {
+              if (success.data.isEmpty) {
                 return const EmptyContactBody();
-              } else {
-                return Padding(
-                  padding: const EdgeInsets.only(left: 8.0, top: 8.0),
-                  child: NoContactsFound(
-                    keyword: controller.searchContactsController.searchKeyword,
-                  ),
-                );
               }
-            }
 
-            return Padding(
-              padding: const EdgeInsets.symmetric(horizontal: 4.0),
-              child: Column(
-                crossAxisAlignment: CrossAxisAlignment.start,
-                children: contactsListSorted
-                    .map<Widget>(
-                      (contact) => InkWell(
-                        borderRadius: BorderRadius.circular(12),
-                        onTap: () {
-                          controller.onContactTap(
-                            context: context,
-                            path: 'contacts',
-                            contact: contact,
+              if (success.data.isEmpty) {
+                if (success.keyword.isEmpty) {
+                  return const EmptyContactBody();
+                } else {
+                  return Padding(
+                    padding: const EdgeInsets.only(left: 8.0, top: 8.0),
+                    child: NoContactsFound(
+                      keyword: success.keyword,
+                    ),
+                  );
+                }
+              }
+
+              return Padding(
+                padding: const EdgeInsets.symmetric(horizontal: 4.0),
+                child: Column(
+                  crossAxisAlignment: CrossAxisAlignment.start,
+                  children: success.data
+                      .map<Widget>(
+                        (contact) => InkWell(
+                          borderRadius: BorderRadius.circular(12),
+                          onTap: () {
+                            controller.onContactTap(
+                              context: context,
+                              path: 'contacts',
+                              contact: contact,
+                            );
+                          },
+                          child: ExpansionContactListTile(contact: contact),
+                        ),
+                      )
+                      .toList()
+                    ..addAll([
+                      ValueListenableBuilder(
+                        valueListenable: controller.isSearchModeNotifier,
+                        builder: (context, isSearchMode, child) {
+                          if (isSearchMode || success.isEnd) {
+                            return const SizedBox.shrink();
+                          }
+                          return const Padding(
+                            padding: EdgeInsets.all(16.0),
+                            child: Center(
+                              child: CircularProgressIndicator(),
+                            ),
```

Comment:
Can you refactor it here?

Reply:
will do in another PR, when use pull_to_refresh